Framework/Geometry/src/Instrument.cpp

```diff
@@ -1235,5 +1237,15 @@ Instrument::ContainsState Instrument::containsRectDetectors() const {
 
 } // containsRectDetectors
 
+/// Only for use by ExperimentInfo. Returns the pointer to the DetectorInfo.
+const Beamline::DetectorInfo *Instrument::detectorInfo() const {
+  return m_detectorInfo;
```

Comment:
Could you add a method to allow the null check? I think `Instrument` had an `isBase` or something like that.